src/Processors/Transforms/SortingTransform.cpp

```diff
@@ -31,6 +31,11 @@ MergeSorter::MergeSorter(Chunks chunks_, SortDescription & description_, size_t
         if (chunk.getNumRows() == 0)
             continue;
 
+        /// Convert to full column, because sparse column has
+        /// access to element in O(log(K)), where K is number of non-default rows,
+        /// which can be inefficient.
+        convertToFullIfSparse(chunk);
```

Comment:
I think it's better to add such comments for each place where we use `convertToFullIfSparse`. Because it looks like code here works with some column internals, but just not ready for sparse format.